ts/Core/Series/Series.ts

```diff
@@ -2572,30 +2476,69 @@ class Series {
     public animate(init?: boolean): void {
         const series = this,
             chart = series.chart,
+            inverted = chart.inverted,
             animation = animObject(series.options.animation),
-            sharedClipKey = this.sharedClipKey;
+            // The key for temporary animation clips
+            animationClipKey = [
+                this.getSharedClipKey(),
+                animation.duration,
+                animation.easing,
+                animation.defer
+            ].join(',');
 
-        // Initialize the animation. Set up the clipping rectangle.
-        if (init) {
+        let animationClipRect = chart.sharedClips[animationClipKey],
+            markerAnimationClipRect = chart.sharedClips[animationClipKey + 'm'];
 
-            series.setClip(animation);
+        // Initialize the animation. Set up the clipping rectangle.
+        if (init && series.group) {
+
+            // Create temporary animation clips
+            if (!animationClipRect) {
+                const clipBox = this.getClipBox();
+                clipBox.width = 0;
+                if (inverted) {
+                    clipBox.x = chart.plotHeight;
+                }
+                animationClipRect = chart.renderer.clipRect(clipBox);
+                chart.sharedClips[animationClipKey] = animationClipRect;
+
+                const markerClipBox = {
+                    // Include the width of the first marker
+                    x: inverted ? (chart.plotSizeX || 0) + 99 : -99,
+                    y: inverted ? -chart.plotLeft : -chart.plotTop,
+                    width: 99,
+                    height: inverted ? chart.chartWidth : chart.chartHeight
+                };
+                markerAnimationClipRect = chart.renderer.clipRect(markerClipBox);
+                chart.sharedClips[animationClipKey + 'm'] = markerAnimationClipRect;
+            }
 
-        // Run the animation
-        } else if (sharedClipKey) {
-            const clipRect = chart.sharedClips[sharedClipKey];
-            const markerClipRect = chart.sharedClips[sharedClipKey + 'm'];
+            series.group.clip(animationClipRect);
+            if (series.markerGroup) {
+                series.markerGroup.clip(markerAnimationClipRect);
+            }
 
-            const finalBox = series.getClipBox(animation, true);
 
-            if (clipRect) {
-                clipRect.animate(finalBox, animation);
-            }
-            if (markerClipRect) {
-                markerClipRect.animate({
-                    width: finalBox.width + 99,
-                    x: finalBox.x - (chart.inverted ? 0 : 99)
-                }, animation);
-            }
+        // Run the animation
+        } else if (
+            animationClipRect &&
+            // Only first series in this pane
+            !animationClipRect.hasClass('highcharts-animating')
+        ) {
+            const finalBox = series.getClipBox();
+            animation.step = (val, fx): void => {
```

Comment:
Q: Is it necessary? In theory, `step` is a valid option in the animation object.

Reply:
Keeping the marker group update in the step of the general group animation will ensure they are in sync, and will probably also be faster because we don't set up extra timers for the marker group. But I'll make sure the custom `animation.step` option is not overwritten.